### PR TITLE
Internal error code

### LINKED ISSUE #6357: [proposal] Define a detailed query internal error code
**Is your feature request related to a problem? Please describe.**
Currently, InternalErrorCode defines some error codes, which are generally placed in ErrorReason and UserException to indicate some internal error causes. If a UserException is thrown during the query, the error code will be returned to the client. For example, `ERROR 1064 (HY000): errCode = 2, detailMessage = Ocurrs time out with specfied time 993 MILLISECONDS`. 1064 is the code of `ErrorCode` and 2 is the InternalErrorCode.

QueryState.ErrType is the result code of the query. There are only two types: Analysis_ERR and OTHER_ERR. The default is OTHER_ERR. When a UserException is caught in `ConnectProcessor.handleQuery` or in `StmtExecutor.execute`, it will be set to Analysis_ERR. OTHER_ERR query will increase MetricRepo.COUNTER_QUERY_ERR.

1. Timeout queries are not counted as failed queries, and the status of the cluster cannot be accurately sensed.
2. When the client gets the query result, it cannot match the error type according to the errCode, and can only perform regular matching according to the `detailMessage`.

**Describe the solution you'd like**
It is recommended to remove QueryState.ErrType and add query-related error codes.

1. Errors caused by users
Syntax error

2. Load related
Timeout (If the non-result node hangs, it may be misjudged as a timeout)
Resource limit

3. Internal system error causes the query to fail
FE sends RPC error to BE
No query node available 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by liutang123**
## Proposed changes

Described in #6357 , define some detailed error code for query.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #6357) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by liutang123** [comment]
@wangbo Do you have time to see this PR? cc @morningman

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java [hunk 1]
```diff
@@ -124,10 +125,19 @@ private void auditAfterExec(String origStmt, StatementBase parsedStmt, Data.PQue
...
-                // err query
-                MetricRepo.COUNTER_QUERY_ERR.increase(1L);
+            if (ctx.getState().getStateType() == QueryState.MysqlStateType.ERR) {
+                InternalErrorCode errorCode = ctx.getState().getInternalErrorCode();
+                if (errorCode == InternalErrorCode.INTERNAL_ERR) {
+                    // err query
+                    MetricRepo.COUNTER_QUERY_ERR.increase(1L);
+                } else {
+                    MetricRepo.COUNTER_QUERY_ERR_OTHER.increase(1L);
+                    MetricRepo.HISTO_QUERY_LATENCY.update(elapseMs);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:135]: Why update `HISTO_QUERY_LATENCY` for this case?

Comment (liutang123) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:135]: Userexception (timeout, cancel, etc) will be treated as `ANALYSIS_ERR`，`ANALYSIS_ERR` will be recorded in the query time. It is understandable that a known error query will be recorded in the query time.
Therefore, this PR still records known errors (timeout, cancel, etc.) in the query time, and other errors (RPC, unknown errors) are not recorded in the query time.

Comment (liutang123) [reply fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java:135]: Maybe we can remove this logic and only record the successful query in the query time

**Review by morningman — COMMENTED** [review]
Please solve the conflict

**Pull request closed without merging**